src/test/regress/expected/with_basics.out

```diff
@@ -220,6 +220,88 @@ JOIN
   users_table USING (user_id);
 ERROR:  cannot push down this subquery
 DETAIL:  Aggregates without group by are currently unsupported
+-- column aliases in CTE
+WITH top_ten(id, val1) AS (
+	SELECT user_id, value_1 FROM users_table ORDER BY value_1 DESC, user_id DESC LIMIT 10
+)
+SELECT * FROM top_ten;
+ id | val1 
+----+------
+  6 |    5
+  6 |    5
+  5 |    5
+  5 |    5
+  5 |    5
+  4 |    5
+  4 |    5
+  3 |    5
+  1 |    5
+  5 |    4
+(10 rows)
+
+-- verify old name is not valid anymore
+WITH top_ten(id, val1) AS (
+	SELECT user_id, value_1 FROM users_table ORDER BY value_1 DESC, user_id DESC LIMIT 10
+)
+SELECT * FROM top_ten ORDER BY user_id DESC;
+ERROR:  column "user_id" does not exist
+LINE 4: SELECT * FROM top_ten ORDER BY user_id DESC;
+                                       ^
+-- verify original name is used if alias is missing
+WITH top_ten(id) AS (
+	SELECT user_id, value_1 FROM users_table ORDER BY value_1 DESC, user_id DESC LIMIT 10
+)
+SELECT * FROM top_ten ORDER BY value_1 DESC;
+ id | value_1 
+----+---------
+  6 |       5
+  6 |       5
+  5 |       5
+  5 |       5
+  5 |       5
+  4 |       5
+  4 |       5
+  3 |       5
+  1 |       5
+  5 |       4
+(10 rows)
+
+-- computed targets from columns also work
+WITH top_ten(id, val, val_mul, val_sum) AS (
+	SELECT user_id, value_1, value_1*2, value_1 + value_2 FROM users_table ORDER BY value_1 DESC, user_id DESC LIMIT 10
+)
+SELECT * FROM top_ten ORDER BY id DESC, val_mul DESC;
```

Comment:
I'd even to some calculations on the final target list such as `SELECT *, id+val, val_mul - 5 FROM top_ten ..` 😄 

Btw, could you please double check that the results are the same with Postgres?

Reply:
final calculations : done,
comparison with postgres results : done